main/lsp/LSPIndexer.cc

```diff
@@ -231,7 +226,24 @@ LSPIndexer::getTypecheckingPath(const LSPFileUpdates &edit,
         prodCategoryCounterInc("lsp.slow_path_reason", "new_file");
         return TypecheckingPath::Slow;
     }
-    return getTypecheckingPathInternal(edit.updatedFiles, evictedFiles);
+
+    LSPFileUpdates::FastPathFilesToTypecheckResult result;
+    auto path = getTypecheckingPathInternal(result, edit.updatedFiles, evictedFiles);
+    if (path == TypecheckingPath::Fast) {
```

Comment:
It would make me feel better if we changed this to a `switch` statement, the reason why we added this enum was in anticipation of adding more than just fast/slow one day (and we're kind of getting closer to that).

Reply:
Great point!